includes/class-multisite-directory-taxonomy.php

```diff
@@ -133,4 +134,30 @@ public function saveTermGeo ($term_id) {
         }
     }
 
+    /**
+     * Adds a dropdown box to filter directory entries by category in the admin listing
+     *
+     * @link http://www.wpcustoms.net/snippets/admin-filter-for-custom-taxonomies/
+     *
+     * @uses $typenow to detect the post type being listed
+     */
+    public function add_taxonomy_admin_filter () {
+
+        global $typenow;
+
+        if ($typenow == Multisite_Directory_Entry::name) {
+            $tax_obj = get_taxonomy(self::name);
+            $tax_name = $tax_obj->labels->name;
+            $terms = get_terms(self::name);
+            if (count($terms) > 0) {
+                echo '<select name="' . self::name . '" id="' . self::name . '" class="postform">';
```

Comment:
Ok! I used `esc_attr()` for attribute values and `esc_html()` for general text, I hope that's correct.